### PR TITLE
updating links from fwlink to real link if exists

### PR DESCRIPTION
**Pull request by eddynaka**
updating links from fwlink to real link if exists - contributes to #3391

### CODE REVIEW

File: docs/framework/windows-workflow-foundation/consuming-odata-feeds-from-a-workflow.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ WCF Data Services is a component of the .NET Framework that enables you to creat
...
+The examples in this topic use the sample Northwind data service located at <https://services.odata.org/Northwind/Northwind.svc/>. This service is provided as part of the [OData SDK](https://www.odata.org/wp-content/uploads/sites/21/odatasdkcodesamples.zip) and provides read-only access to the sample Northwind database. If write access is desired, or if a local WCF Data Service is desired, you can follow the steps of the [WCF Data Services Quickstart](https://docs.microsoft.com/dotnet/framework/data/wcf/quickstart-wcf-data-services) to create a local OData service that provides access to the Northwind database. If you follow the quickstart, substitute the local URI for the one provided in the example code in this topic.
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/consuming-odata-feeds-from-a-workflow.md]:
```suggestion
The examples in this topic use the sample Northwind data service located at <https://services.odata.org/Northwind/Northwind.svc/>. This service is provided as part of the [OData SDK](https://www.odata.org/wp-content/uploads/sites/21/odatasdkcodesamples.zip) and provides read-only access to the sample Northwind database. If write access is desired, or if a local WCF Data Service is desired, you can follow the steps of the [WCF Data Services Quickstart](../data/wcf/quickstart-wcf-data-services.md) to create a local OData service that provides access to the Northwind database. If you follow the quickstart, substitute the local URI for the one provided in the example code in this topic.
```

File: docs/framework/windows-workflow-foundation/dynamic-update.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ DynamicUpdateServices.PrepareForUpdate(ab);
 ```
 
 > [!NOTE]
-> To download the sample code that accompanies this topic, see [Dynamic Update sample code](https://go.microsoft.com/fwlink/?LinkId=227905).
+> To download the sample code that accompanies this topic, see [Dynamic Update sample code](https://docs.microsoft.com/samples/browse/-samples).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/dynamic-update.md]: let's undo this change. this is not the right target for this. we'd have to find the source.

File: docs/framework/windows-workflow-foundation/feature-specifics.md [hunk 3]
```diff
@@ -90,15 +90,15 @@ The WCF configuration schema is complex and provides users with many hard to fin
 
 ### Getting Started
 
-- [A Developer's Guide to WCF 4.0](https://go.microsoft.com/fwlink/?LinkId=204940)
+- [A Developer's Guide to WCF 4.0](https://docs.microsoft.com/previous-versions/dotnet/articles/ee354381(v=msdn.10))
 
-- [Configuration Channel Factory](https://go.microsoft.com/fwlink/?LinkId=204941)
+- [Configuration Channel Factory](https://docs.microsoft.com/dotnet/api/system.servicemodel.configuration.configurationchannelfactory-1&view=netframework-4.8)
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/feature-specifics.md]:
```suggestion
- [Configuration Channel Factory](xref:System.ServiceModel.Configuration.ConfigurationChannelFactory%601)
```

File: docs/framework/windows-workflow-foundation/feature-specifics.md [hunk 4]
```diff
@@ -90,15 +90,15 @@ The WCF configuration schema is complex and provides users with many hard to fin
...
 
-- [Standard Endpoint Element](https://go.microsoft.com/fwlink/?LinkId=204942)
+- [Standard Endpoint Element](https://docs.microsoft.com/dotnet/api/system.servicemodel.configuration.standardendpointelement&view=netframework-4.8)
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/feature-specifics.md]:
```suggestion
- [Standard Endpoint Element](xref:System.ServiceModel.Configuration.StandardEndpointElement)
```

File: docs/framework/windows-workflow-foundation/feature-specifics.md [hunk 5]
```diff
@@ -120,9 +120,9 @@ The [DataContractResolver](../wcf/samples/datacontractresolver.md) solves these
 
 ### Getting Started
 
-- [Data Contract Resolver API documentation](https://go.microsoft.com/fwlink/?LinkId=204946)
+- [Data Contract Resolver API documentation](https://docs.microsoft.com/dotnet/api/system.runtime.serialization.datacontractresolver&view=netframework-4.8)
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/feature-specifics.md]:
```suggestion
- [Data Contract Resolver API documentation](xref:System.Runtime.Serialization.DataContractResolver)
```

File: docs/framework/windows-workflow-foundation/how-to-create-a-custom-tracking-participant.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ ms.assetid: 1b612c7e-2381-4a7c-b07a-77030415f2a3
...
   
 > [!NOTE]
-> Each topic in the Getting Started tutorial depends on the previous topics. To complete this topic, you must first complete the previous topics. To download a completed version or view a video walkthrough of the tutorial, see [Windows Workflow Foundation (WF45) - Getting Started Tutorial](https://go.microsoft.com/fwlink/?LinkID=248976).  
+> Each topic in the Getting Started tutorial depends on the previous topics. To complete this topic, you must first complete the previous topics. To download a completed version or view a video walkthrough of the tutorial, see [Windows Workflow Foundation (WF45) - Getting Started Tutorial](https://docs.microsoft.com/samples/browse/-samples).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/how-to-create-a-custom-tracking-participant.md]: We'll have to undo all of these that are going to the sample browser

File: docs/framework/windows-workflow-foundation/how-to-enable-sql-persistence-for-workflows-and-workflow-services.md [hunk 7]
```diff
@@ -147,7 +147,7 @@ workflowServiceHost.DurableInstancingOptions.InstanceStore = sqlInstanceStoreObj
 ```
 
 > [!IMPORTANT]
-> It is recommended that you do not store sensitive information such as user names and passwords in the Web.config file. If you do store sensitive information in the Web.config file, you should secure access to the Web.config file by using file system Access Control Lists (ACLs). In addition, you can also secure the configuration values within a configuration file as mentioned in [Encrypting Configuration Information Using Protected Configuration](https://go.microsoft.com/fwlink/?LinkId=178419).
+> It is recommended that you do not store sensitive information such as user names and passwords in the Web.config file. If you do store sensitive information in the Web.config file, you should secure access to the Web.config file by using file system Access Control Lists (ACLs). In addition, you can also secure the configuration values within a configuration file as mentioned in [Encrypting Configuration Information Using Protected Configuration](https://docs.microsoft.com/previous-versions/53tyfkaw(v=vs.140)).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/how-to-enable-sql-persistence-for-workflows-and-workflow-services.md]:
```suggestion
> It is recommended that you do not store sensitive information such as user names and passwords in the Web.config file. If you do store sensitive information in the Web.config file, you should secure access to the Web.config file by using file system Access Control Lists (ACLs). In addition, you can also secure the configuration values within a configuration file as mentioned in [Encrypting Configuration Information Using Protected Configuration](https://docs.microsoft.com/en-us/previous-versions/aspnet/53tyfkaw(v=vs.100)).
```

File: docs/framework/windows-workflow-foundation/performance.md [hunk 8]
```diff
@@ -11,7 +11,7 @@ ms.assetid: 67d2b3e8-3777-49f8-9084-abbb33b5a766
...
+ The version of [!INCLUDE[wf1](../../../includes/wf1-md.md)] introduced in .NET Framework 4 will be referred to as WF4 for the rest of this topic. [!INCLUDE[wf1](../../../includes/wf1-md.md)] was introduced in .NET Framework 3.0 and had a few minor revisions through .NET Framework 3.5 SP1. The .NET Framework 3.5 version of Workflow Foundation will be referred to as WF3 for the rest of this topic. WF3 is shipped in .NET Framework 4 side-by-side with WF4. For more information about migrating WF3 artifacts to WF4 see: [Windows Workflow Foundation 4 Migration Guide](https://docs.microsoft.com/dotnet/framework/windows-workflow-foundation/migration-guidance).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/performance.md]:
```suggestion
 The version of [!INCLUDE[wf1](../../../includes/wf1-md.md)] introduced in .NET Framework 4 will be referred to as WF4 for the rest of this topic. [!INCLUDE[wf1](../../../includes/wf1-md.md)] was introduced in .NET Framework 3.0 and had a few minor revisions through .NET Framework 3.5 SP1. The .NET Framework 3.5 version of Workflow Foundation will be referred to as WF3 for the rest of this topic. WF3 is shipped in .NET Framework 4 side-by-side with WF4. For more information about migrating WF3 artifacts to WF4 see: [Windows Workflow Foundation 4 Migration Guide](migration-guidance.md).
```

File: docs/framework/windows-workflow-foundation/performance.md [hunk 9]
```diff
@@ -295,7 +295,7 @@ The following diagram shows the basic compensation workflow. The WF3 workflow is
...
+ Measuring memory per workflow definition is divided into two different tests because of the available options for hosting workflows in WF3 and WF4.  The tests are run in a different manner than the workflow complexity test in that a given workflow is instanced and executed only once per definition.  This is because the workflow definition and its host remain in memory for the lifetime of the AppDomain.  The memory used by running a given workflow instance should be cleaned up during garbage collection.  The migration guidance for WF4 contains more detailed information on the hosting options. For more information, see [WF Migration Cookbook: Workflow Hosting](https://docs.microsoft.com/dotnet/framework/windows-workflow-foundation/migration-guidance).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/performance.md]:
```suggestion
 Measuring memory per workflow definition is divided into two different tests because of the available options for hosting workflows in WF3 and WF4.  The tests are run in a different manner than the workflow complexity test in that a given workflow is instanced and executed only once per definition.  This is because the workflow definition and its host remain in memory for the lifetime of the AppDomain.  The memory used by running a given workflow instance should be cleaned up during garbage collection.  The migration guidance for WF4 contains more detailed information on the hosting options. For more information, see [WF Migration Cookbook: Workflow Hosting](migration-guidance.md).
```

File: docs/framework/windows-workflow-foundation/performance.md [hunk 10]
```diff
@@ -428,7 +428,7 @@ public class Workflow1 : Activity
...
+ WF4 is almost a complete rewrite of [!INCLUDE[wf1](../../../includes/wf1-md.md)] and therefore WF3 workflows and activities are not directly compatible with WF4.  Many customers that adopted Windows Workflow Foundation early will have in-house or third-party workflow definitions and custom activities for WF3.  One way to ease the transition to WF4 is to use the Interop activity, which can execute WF3 activities from within a WF4 workflow.  It is recommended that the <xref:System.Activities.Statements.Interop> activity only be used when necessary. For more information about migrating to WF4 check out the [WF4 Migration Guidance](https://docs.microsoft.com/dotnet/framework/windows-workflow-foundation/migration-guidance).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/performance.md]:
```suggestion
 WF4 is almost a complete rewrite of [!INCLUDE[wf1](../../../includes/wf1-md.md)] and therefore WF3 workflows and activities are not directly compatible with WF4.  Many customers that adopted Windows Workflow Foundation early will have in-house or third-party workflow definitions and custom activities for WF3.  One way to ease the transition to WF4 is to use the Interop activity, which can execute WF3 activities from within a WF4 workflow.  It is recommended that the <xref:System.Activities.Statements.Interop> activity only be used when necessary. For more information about migrating to WF4 check out the [WF4 Migration Guidance](migration-guidance.md).
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @eddynaka. Left a few comments for you to consider before we can merge.

File: docs/framework/windows-workflow-foundation/dynamic-update.md (hunk 2 above)
Comment (eddynaka) [reply docs/framework/windows-workflow-foundation/dynamic-update.md]: ok!

File: docs/framework/windows-workflow-foundation/how-to-create-a-custom-tracking-participant.md (hunk 6 above)
Comment (eddynaka) [reply docs/framework/windows-workflow-foundation/how-to-create-a-custom-tracking-participant.md]: ok!

File: docs/framework/windows-workflow-foundation/how-to-host-multiple-versions-of-a-workflow-side-by-side.md [hunk 11]
```diff
@@ -11,7 +11,7 @@ ms.assetid: 09c575df-e0a3-4f3b-9e01-a7ac59d65287
 `WorkflowIdentity` provides a way for workflow application developers to associate a name and a version with a workflow definition, and for this information to be associated with a persisted workflow instance. This identity information can be used by workflow application developers to enable scenarios such as side-by-side execution of multiple versions of a workflow definition, and provides the cornerstone for other functionality such as dynamic update. This step in the tutorial demonstrates how to use `WorkflowIdentity` to host multiple versions of a workflow at the same time.
 
 > [!NOTE]
-> To download a completed version or view a video walkthrough of the tutorial, see [Windows Workflow Foundation (WF45) - Getting Started Tutorial](https://go.microsoft.com/fwlink/?LinkID=248976).
+> To download a completed version or view a video walkthrough of the tutorial, see [Windows Workflow Foundation (WF45) - Getting Started Tutorial](https://docs.microsoft.com/samples/browse/-samples).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/how-to-host-multiple-versions-of-a-workflow-side-by-side.md]:
```suggestion
> To download a completed version or view a video walkthrough of the tutorial, see [Windows Workflow Foundation (WF45) - Getting Started Tutorial](https://go.microsoft.com/fwlink/?LinkID=248976).
```

**Review by mairaw — COMMENTED** [review]
One more to roll back

**Review by mairaw — APPROVED** [review]
:shipit: thanks!
